### PR TITLE
use ws2_32 posix like function signatures for std.c

### LINKED ISSUE #9971: sendto is declared as coming from libc on Windows instead of ws2_32
### Steps to Reproduce

bug.zig:
```zig
const std = @import("std");

pub fn main() !void {
    const msg = &[_]u8 { 0 };
    const send_addr = std.net.Address.parseIp4("255.255.255.255", 1234) catch unreachable;    
    _ = try std.os.sendto(std.os.windows.ws2_32.INVALID_SOCKET, msg, 0, &send_addr.any, send_addr.getOsSockLen());
}
```
```batch
> zig build-exe bug.zig
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5260:26: error: dependency on libc must be explicitly specified in the build command
        const rc = system.sendto(sockfd, buf.ptr, buf.len, flags, dest_addr, addrlen);
                         ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5262:37: note: referenced here
            if (rc == windows.ws2_32.SOCKET_ERROR) {
                                    ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os\windows\ws2_32.zig:1845:51: note: referenced here
pub extern "ws2_32" fn WSAGetLastError() callconv(WINAPI) WinsockError;
                                                  ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os\windows\ws2_32.zig:1845:59: note: referenced here
pub extern "ws2_32" fn WSAGetLastError() callconv(WINAPI) WinsockError;
                                                          ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5263:39: note: referenced here
                switch (windows.ws2_32.WSAGetLastError()) {
                                      ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5283:49: note: referenced here
                    else => |err| return windows.unexpectedWSAError(err),
                                                ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5258:21: note: referenced here
) SendToError!usize {
```

### Expected Behavior

The example should compile.  This example works on zig version 0.8.1.  I believe this was broken by this commit: 7884d84315f4080bfe429d3e29f7ebedf9f3f6  in the file `lib/std/os.zig` with the following change:

```patch
/// When not linking libc, it is the OS-specific system interface.
 pub const system = if (@hasDecl(root, "os") and root.os != @This())
     root.os.system
-else if (builtin.link_libc)
+else if (builtin.link_libc or is_windows)
    std.c
 else switch (builtin.os.tag) {
```

`std.os.sendto` calls `system.sendto` and this change routed `system.sendto` to `std.c` instead of `std.os.windows` where the correct declaration exists on windows.  So this is probably a problem for alot more functions, but this is the first one I found and noticed.  I don't quite understand why this change was made so I don't know whether the fix is to undo this particular line of the commit or if there's more to what's going on here.

### Actual Behavior

Zig thinks `sendto` is defined in `libc` on Windows instead of `ws2_32` where it actually lives.

### LINKED ISSUE #10934: Windows std.c incorrect function prototypes
### Zig Version

0.10.0-dev.856+e86a89d3f

### Steps to Reproduce

When using `std.c` on Window, there are functions externed using Unix prototypes such as `recvfrom` which causes unexpected behavior because the Windows function prototypes are different. Also, functions like `getdirentries` are present when compiling on Windows even though they are not present on Windows which causes an undefined reference when attempting to use them.

For example, the Unix prototype of `recvfrom` is the following:
```c
ssize_t recvfrom(int socket, void *restrict buffer, size_t length,
       int flags, struct sockaddr *restrict address,
       socklen_t *restrict address_len);
```
However, on Windows the prototype is this instead:
```c
int recvfrom(
  [in]                SOCKET   s,
  [out]               char     *buf,
  [in]                int      len,
  [in]                int      flags,
  [out]               sockaddr *from,
  [in, out, optional] int      *fromlen
);
```

Because the return types are not the same and the functions are `extern`'d, the below behavior is observed.

### Expected Behavior

```zig
const std = @import("std");

pub fn main() void {
    std.log.info("{}", .{std.c.getdirentries});
}
```
I would expect the above program to output the following error
```
.\src\main.zig:4:33: error: container 'std.c' has no member called 'getdirentries'
    std.log.info("{}", .{std.c.getdirentries});
```

And I would expect the below code to receive data from the socket and not return an error. As far as I know there shouldn't be anything wrong this example.
```zig
const std = @import("std");
const os = std.os;

pub fn main() !void {
    const sock = try os.socket(os.AF.INET, os.SOCK.DGRAM, 0);
    defer os.closeSocket(sock);

    const address = std.net.Address.initIp4([_]u8{ 127, 0, 0, 1 }, 4096);
    try os.bind(sock, &address.any, address.getOsSockLen());

    var buffer: [std.mem.page_size]u8 = undefined;
    var sockaddr: os.sockaddr = undefined;
    var len: os.socklen_t = undefined;
    const read_len = try os.recvfrom(sock, &buffer, 0, &sockaddr, &len);
    std.log.info("read_len = {}", .{read_len});
}
```

### Actual Behavior

The first example from above has the following compiler error
```
lld-link: error: undefined symbol: getdirentries
```
and the second example yields the following output
```
info: read_len = 4294967295
```

Due to the difference in the return types of `recvfrom` on Windows and Unix platforms, and due to #9971 meaning that `std.os.recvfrom` calls `std.c.recvfrom`, the return value of `std.c.recvfrom` is interpreted as an `isize` when it is actually a `c_int` and causes `4294967295` to be returned from the function instead of `-1`.

Also, calling `std.os.windows.ws2_32.WSAGetLastError()` afterwards indicates that the error that occurred was `WSAEFAULT` which indicates the [following](https://docs.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2):
> Bad address.
    The system detected an invalid pointer address in attempting to use a pointer argument of a call. This error occurs if an application passes an invalid pointer value, or if the length of the buffer is too small. For instance, if the length of an argument, which is a sockaddr structure, is smaller than the sizeof(sockaddr).

I believe this means that there is also an issue with how the socket types are defined on Windows as well.

### PR DESCRIPTION
**Pull request by arBmind**
Fixes #9971

* fixes issues with different size types

### CODE REVIEW

**Review by andrewrk — CHANGES_REQUESTED** [review]
Thanks for the patch. I have one request before merging-

File: lib/std/os/windows/ws2_32.zig [hunk 1]
```diff
@@ -1701,137 +1701,166 @@ pub const WinsockError = enum(u16) {
...
-    optname: i32,
-    optval: ?[*]const u8,
-    optlen: i32,
-) callconv(WINAPI) i32;
-
-pub extern "ws2_32" fn shutdown(
-    s: SOCKET,
-    how: i32,
-) callconv(WINAPI) i32;
+pub const posixLike = struct {
```
[the remark this answers is not in the document]
Comment (arBmind) [reply lib/std/os/windows/ws2_32.zig]: Okay, to do that I copied all the posix like functions of ws2_32 api to c.zig and changed the formatting a match the c equivalents.

Also realized that I moved too many functions to posixLike (as they were mentioned in winsock documentation) that are not part of c.zig.

**Comment by andrewrk** [comment]
std.c is for the C library, not for ws2_32. Please see #15223 for a resolution of this issue. The std.os code should be modified to call std.windows.ws2_32 on windows rather than going through std.c.

**Comment by andrewrk** [comment]
Sendto issue addressed in #15225.

**Pull request closed without merging**